packages/v/volk/xmake.lua

```diff
@@ -0,0 +1,24 @@
+package("volk")
+
+    set_homepage("https://github.com/zeux/volk")
+    set_description("volk is a meta-loader for Vulkan")
+    set_license("MIT")
+
+    add_urls("https://github.com/zeux/volk/archive/$(version).tar.gz")
+    add_versions("1.2.162", "ac4d9d6e88dee5a83ad176e2da57f1989ca2c6df155a0aeb5e18e9471aa4d777")
+
+    add_deps("cmake")
+
+    on_install("windows", "linux", "macosx", function (package)
+        import("package.tools.cmake").install(package)
+    end)
+
+    on_test(function (package)
+        assert(package:check_cxxsnippets({test = [[
+            void test() {
+                volkInitialize();
+            }
+            }
+        ]]}, {configs = {languages = "c++14"},
+              includes = {"volk.h"}}))
```

Comment:
We need only write 

```lua
        assert(package:has_cfuncs("volkInitialize", {configs = {languages = "c++14"}, includes = "volk.h"}))
```

Reply:
@waruqi 
Hello , I update this and get CI  test error.
 >/home/runner/.xmake/cache/packages/2011/v/volk/1.2.162/source/volk-1.2.162/volk.h:50:12: fatal error: vulkan/vulkan.h: No such file or directory

But I add deps vulkan-headers before
>finding vulkan-headers from xmake ..
{ 
  sysincludedirs = { 
    "/home/runner/.xmake/packages/v/vulkan-headers/v1.2.162/531f88cff3a64da0aa3fb05ee281ab7d/include", 
  },
  license = "Apache-2.0",
  version = "v1.2.162", 
}

> /usr/bin/gcc -c -m64 -isystem /home/runner/.xmake/packages/v/vulkan-headers/v1.2.162/531f88cff3a64da0aa3fb05ee281ab7d/include -o /dev/shm/.xmake1001/201130/_12FAB8FFF1004010869BB67B15884E20.o /dev/shm/.xmake1001/201130/_4819AB359CBC4A008C959CC1F587B110.c
> checking for c includes(vulkan/vulkan.h)
> checking for c snippet(test)
  => install vulkan-headers v1.2.162 .. ok
  => download https://github.com/zeux/volk/archive/1.2.162.tar.gz .. ok


Is there something need to change?

Reply:
cmake failed to detect volkan headers, we can only manually pass it to cmake cflags through packagedeps.

```lua
    on_install("windows", "linux", "macosx", function (package)
        local configs = {}
        if package:config("shared") then
            table.insert(configs, "--enable-shared=yes")
        else
            table.insert(configs, "--enable-shared=no")
        end
        import("package.tools.cmake").build(package, configs, {buildir = "build", packagedeps = "vulkan-headers"})
        if package:is_plat("windows") then
            os.trycp("build/*.lib", package:installdir("lib"))
            os.trycp("build/*.dll", package:installdir("bin"))
        else
            os.trycp("build/*.a", package:installdir("lib"))
            os.trycp("build/*.so", package:installdir("lib"))
        end
        os.cp("*.h", package:installdir("include"))
    end)
```